src/form/Form.tsx

```diff
@@ -999,16 +1019,18 @@ function Form({
       });
       if (invalid) return;
 
+      submitHidden();
       // async execution after user's onSubmit
       return submitAndNavigate({
         metadata,
         formattedFields
       });
     } else {
-      return submitAndNavigate({
-        metadata,
-        formattedFields
-      });
+      submitHidden();
+      // return submitAndNavigate({
```

Comment:
please remove commented out code or uncomment

Reply:
this kinda seems like it should maybe just be int he submitAndNavigate function if both functions need to be c alled in both places